goals.md

```diff
@@ -0,0 +1,61 @@
+
+This specification will be independent of the programming language its Components are written in, so it will be possible to mix-and-match.
+Instead of implementing all the necessary Components, we want to give the communication tools to connect everything, to balance on the shoulders of giants.
+
+## Small to medium-sized use cases
+We want ECP to serve laboratory setups of up to approximately 50 participants on one or more computation Nodes on a common network.
+For example, you might have up to ~dozens of instruments, a few data acquisition nodes, one or several GUIs and control nodes.
+
+## Flexible and modular
+"LEGO"-like modularity is important. We want to reassemble, change the constellation of devices, easily adapt to changing experiments, data rate needs, etc.
+
+We want people to easily be able to write their own software implementing (parts of) this protocol, for example to create just the GUI tool that they need.
+
+We want ECP to be very lightly coupled to the used software (like instrument driver libraries) -- we want to thinly wrap these with some interface code that is written (ideally) _once_.
+Given some instrument control library with a reasonably homogeneous/simple API, it should be easy to write an Actor variant to connect that library's Drivers with our Actor interface.
+
+## Simple, yet powerful
+You can send a message to any Component by specifying its (possibly human-readable) name or ID, similar to email.
+
+We will have a default interpretation of the message content to implement remote procedure call (RPC), i.e. to set/read properties or call methods.
+
+## Fast yet reliable
+We want to communicate asynchronously and concurrently: The Director might want to send several commands to different devices at the same time, without waiting for the answer of a (slow) instrument before writing to the next one.
+This way, communication delays will be determined by the slowest instrument, not by the number of instruments.
+
+Messages should be reliably delivered by the protocol and not lost (barring network problems, crashing Components, hardware failure, etc.).
+
+A failing node should not corrupt the measurement data acquired until then.
+
+## Onboarding experience
+We want to leave the entry threshold as low as possible, the learning curve flat, and the usability of ECP modular.
+
+For simple measurements, a user should just need to download a package, write a little code to connect to the hardware with an Actor and a Driver from some instrument control library (e.g. a `pymeasure.Instrument`), and measure some things using the Actor interface.
+This assumes that an Actor variant interfacing to the given instrument control library has already been implemented.
+
+It should be possible to slowly scale up -- start with a single instrument that you exchange messages with, then add more Actors, a data sink, and a Director for control.
```

Comment:
In my understanding, the minimum requirement is an Actor **and** a Director, as the script controlling the Actor is already a (not feature complete) Director.

However, the idea is clear and we can revisit this page, when things are clearer.

Reply:
I think, it should be possible to have (an) Actor(s) act on messages received. If they are correctly structured, it does not matter where from.
It could also be a Processor, e.g. a PID holding a temperature, does not have to be a Director. 

I see your point though, that a program controlling the actor "quacks" like a Director already (but then: Processors).
Maybe our Minimal Viable Director is a class sending messages out on certain calls.

I agree, we can leave it now and refine it when we hash out the Director API in more detail.